proto/kv/v1/kv.proto

```diff
@@ -26,12 +26,22 @@ service KeyValue {
   rpc Query (KeyValueQueryRequest) returns (KeyValueQueryResponse);
 }
 
+// Key type for KV Store
+message Key {
+  oneof key {
+    // Value for a string key
+    string string = 1;
+    // Value for a number key
+    int number = 2;
```

Comment:
Happy to leave the number type off for now and stick to strings only. This change just allows us to extend key capability in future.